tests/conftest.py

```diff
@@ -188,3 +190,75 @@ def api_client_with_credentials(db, token_user, api_client):
 @pytest.fixture
 def admin_site():
     return AdminSite()
+
+
+@pytest.fixture
+def moneygram_fsp():
+    return FinancialServiceProviderFactory(vendor_number="MONEYGRAM")
+
+
+@pytest.fixture
+def moneygram_record(moneygram_fsp):
+    return PaymentRecordFactory(parent__fsp=moneygram_fsp, fsp_code="TEST123")
+
+
+@pytest.fixture
+def request_factory():
+    return RequestFactory()
+
+
+@pytest.fixture
+def request_with_messages(request_factory):
+    request = request_factory.post("/")
+    middleware = SessionMiddleware(lambda r: None)
+    middleware.process_request(request)
+    middleware = MessageMiddleware(lambda r: None)
+    middleware.process_request(request)
+    return request
+
+
+@pytest.fixture
+def sample_queryset():
+    from hope_payment_gateway.apps.gateway.models import PaymentRecord
+
+    return PaymentRecord.objects.all()
+
+
+@pytest.fixture
+def modeladmin():
+    from django.contrib.admin import ModelAdmin
+    from hope_payment_gateway.apps.gateway.models import PaymentRecord
+    from unittest.mock import MagicMock
+
+    admin = MagicMock(spec=ModelAdmin)
+    admin.model = PaymentRecord
+    admin.opts = MagicMock()
+    admin.opts.verbose_name_plural = "Payment Records"
+    admin.opts.app_label = "gateway"
+    admin.admin_site = MagicMock()
+    admin.admin_site.name = "admin"
+    admin.admin_site.each_context.return_value = {}
+    admin.get_fieldsets.return_value = []
+    return admin
+
+
+@pytest.fixture
+def request_with_data(request_with_messages, admin_user):
```

Comment:
maybe this doesn't need to be used in the global conftest (we put here only reusable things)